modules/client/src/main/java/org/apache/ignite/internal/client/io/netty/NettyClientConnectionMultiplexer.java

```diff
@@ -75,6 +90,74 @@ public void initChannel(SocketChannel ch) {
         }
     }
 
+    private void setupSsl(SocketChannel ch, IgniteClientConfiguration clientCfg) {
+        if (clientCfg.sslConfiguration() == null || !clientCfg.sslConfiguration().enabled()) {
+            return;
+        }
+
+        try {
+            var trustStore = clientCfg.sslConfiguration().trustStore();
+            if (trustStore == null) {
+                throw new IgniteClientSslException("SSL is enabled but no trust store is provided");
+            }
+            if (trustStore.path() == null) {
+                throw new IgniteClientSslException("SSL is enabled but no path to trust store is provided");
+            }
```

Comment:
None of the properties in the client-side `SslConfiguration` should be mandatory. We should be able to use SSL without providing anything (for example, you don't need a key store or a trust store to load `https://github.com`).